### PR TITLE
CDHSH-87 Added frontend edit dialog for personal details on the profile

### PR DESCRIPTION
**Pull request by Liannus**
Changelog:
- Added dialog to edit the personal details on the profile page

Implementation Details:
- Created controlled components to easily compile data for updating

### CODE REVIEW

File: services/frontend/src/scenes/Profile/ProfileLayout.js [hunk 1]
```diff
@@ -86,32 +88,58 @@ const renderSectionComponent = (sectionName, sectionProps) => {
...
-                    {profile.phone}
-                </p>
-            </div>
-        </div>
-    </Paper>
-);
+const PersonalDetails = ({profile}) => {
+    const [personalDetailsDialogOpen, setPersonalDetailsDialogOpen] = React.useState(false);
+
+    function openDialog() {
```
Comment (DevinSit) [inline services/frontend/src/scenes/Profile/ProfileLayout.js]: Are you purposely using regular functions just to check if I'm actually reviewing your code or something?...

File: services/frontend/src/scenes/Profile/ProfileLayout.js [hunk 2]
```diff
@@ -86,32 +88,58 @@ const renderSectionComponent = (sectionName, sectionProps) => {
...
-                <p className="profile-card-text">
-                    <LocalPhone />
-                    {profile.phone}
-                </p>
-            </div>
-        </div>
-    </Paper>
-);
+const PersonalDetails = ({profile}) => {
+    const [personalDetailsDialogOpen, setPersonalDetailsDialogOpen] = React.useState(false);
```
Comment (DevinSit) [inline services/frontend/src/scenes/Profile/ProfileLayout.js]: I enjoy that this file now mixes using hooks directly from the `React` object _as well as_ importing them directly from the package.

**Comment by TunaButter** [comment]
> The things I could complain about are not worth complaining about, so you get some only slightly-more-worth-complaining-about comments that are borderline not-worth-complaining-about.

You're an A-hole. Complain about this, b**ch.

**Comment by DevinSit** [comment]
> > The things I could complain about are not worth complaining about, so you get some only slightly-more-worth-complaining-about comments that are borderline not-worth-complaining-about.
> 
> You're an A-hole. Complain about this, b**ch.

_reported_ :stuck_out_tongue:

**Comment by TunaButter** [comment]
> > > The things I could complain about are not worth complaining about, so you get some only slightly-more-worth-complaining-about comments that are borderline not-worth-complaining-about.
> > 
> > 
> > You're an A-hole. Complain about this, b**ch.
> 
> _reported_ 😛

There is no middle finger emoji in the reactions.

**Comment by DevinSit** [comment]
> > > > The things I could complain about are not worth complaining about, so you get some only slightly-more-worth-complaining-about comments that are borderline not-worth-complaining-about.
> > > 
> > > 
> > > You're an A-hole. Complain about this, b**ch.
> > 
> > 
> > _reported_ stuck_out_tongue
> 
> There is no middle finger emoji in the reactions.

Shame, damn shame...

File: services/frontend/src/scenes/Profile/ProfileLayout.js (hunk 1 above)
Comment (Liannus) [reply services/frontend/src/scenes/Profile/ProfileLayout.js]: Totally,

Swapped to arrow functions.

File: services/frontend/src/scenes/Profile/ProfileLayout.js (hunk 2 above)
Comment (Liannus) [reply services/frontend/src/scenes/Profile/ProfileLayout.js]: That's true and something I didn't notice.

Loaded useState in our imports.

**Comment by Liannus** [comment]
👍 11/10 feedback right there

**Pull request merged**